Documentation/ExtensionArchitecture/Extbase/Reference/_FileUpload/_BlogEnhanced.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\Domain\Model;
+
+use TYPO3\CMS\Core\Resource\Enum\DuplicationBehavior;
+use TYPO3\CMS\Extbase\Annotation\FileUpload;
+use TYPO3\CMS\Extbase\Domain\Model\FileReference;
+use TYPO3\CMS\Extbase\DomainObject\AbstractEntity;
+use TYPO3\CMS\Extbase\Persistence\ObjectStorage;
+
+class Blog extends AbstractEntity
+{
+    // A single file
+    #[FileUpload([
+        'validation' => [
+            'required' => true,
+            'maxFiles' => 1,
+            'fileSize' => ['minimum' => '0K', 'maximum' => '2M'],
+            'allowedMimeTypes' => ['image/jpeg'],
+            'imageDimensions' => ['maxWidth' => 4096, 'maxHeight' => 4096],
+        ],
+        'uploadFolder' => '1:/user_upload/extbase_single_file/',
```

Comment:
IMO this path has to be created first on your own.
This is just a note for me. I have not read everything until now.

Reply:
There is a `createUploadFolderIfNotExist` attribute, which defines, how this is handled. It defaults to `true`, so none existing folders will automatically be created.

Reply:
@derhansen https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/4808/files#r1796675826
:D :D